src/python/py/models/builder.py

```diff
@@ -2447,6 +2577,38 @@ def __init__(self, config, io_dtype, onnx_dtype, ep, cache_dir, extra_options):
         super().__init__(config, io_dtype, onnx_dtype, ep, cache_dir, extra_options)
 
 
+class Phi3MoE128KModel(MistralModel):
+    def __init__(self, config, io_dtype, onnx_dtype, ep, cache_dir, extra_options):
+        super().__init__(config, io_dtype, onnx_dtype, ep, cache_dir, extra_options)
+        assert io_dtype == TensorProto.FLOAT16, "This model only supports float16 io type."
+
+        self.rotemb_attrs["rotemb_name"] = "rotary_emb"
+
+        self.layernorm_attrs["simple"] = False
+
+        self.moe_attrs["num_experts"] = 16
+        self.moe_attrs["top_k"] = 2
+        self.moe_attrs["activation_type"] = "silu"
```

Comment:
Once these are auto-added when `moe_attrs` is created, I think these entries can be removed.